src/assert.js

```diff
@@ -65,16 +74,25 @@ function assertion(errorCls, shouldBeTruthy, opt_message, var_args) {
     return shouldBeTruthy;
   }
 
+  // Skip the first 3 arguments to isolate format params
+  const messageArgs = Array.prototype.slice.call(arguments, 3);
   // Substitute provided values into format string in message
-  const message = Array.prototype.slice
-    // Skip the first 3 arguments to isolate format params
-    .call(arguments, 3)
-    .reduce(
-      (msg, subValue) => msg.replace('%s', subValue),
-      opt_message || 'Assertion failed'
-    );
+  const message = messageArgs.reduce(
+    (msg, subValue) => msg.replace('%s', subValue),
+    opt_message || 'Assertion failed'
+  );
+  const error = new errorCls(message);
+  error.messageArray = messageArgs;
+
+  // If an element is provided, add it to the error object
+  for (let i = 0; i < messageArgs.length; ++i) {
+    if (messageArgs[i]?.tagName) {
```

Comment:
Nit: This will be faster

```suggestion
    if (messageArgs[i]?.nodeType === 1) {
```

Reply:
Updated